cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -40,11 +40,40 @@ type MasterConfiguration struct {
 	KubernetesVersion string
 	CloudProvider     string
 	AuthorizationMode string
+
+	Phases Phases
+
+	// The directory where certificates are stored
+	CertificatesDir string
+}
+
+type PhaseMetadata struct {
```

Comment:
I don't plan to add anything more than annotations, but I think there must be some place to put unstructured data, if it's really needed (e.g. we have lack of oversight for something, we can use the annotations as a last resort to put data. Note: only as a last resort.).
Also, I expect that the user might want to put extra data here, as we've said earlier, we don't know how this will actually be used in the wild.